app.js

```diff
@@ -0,0 +1,70 @@
+(function() {
+  const todos = ["Walk the dog", "Go shopping for new Jeans", "Learn web"];
+  const inputEl = document.getElementById("todo-input");
+  const addBtnEl = document.getElementById("add-btn");
+  const todoListEl = document.getElementById("todo-list");
+  let newTodo = "";
+
+  // Add functionality to HTML elements
+  inputEl.addEventListener("keyup", event => (newTodo = event.target.value));
+  addBtnEl.addEventListener("click", () => {
+    if (newTodo.length > 0) {
+      addTodo(newTodo);
+      newTodo = "";
+      inputEl.value = "";
+    }
+  });
+
+  // Add existing todos to the DOM
+  todos.forEach(todo => addTodo(todo));
+
+  function addTodo(todo) {
+    const todoEl = getTodoItemElement(todo);
+    todoListEl.insertBefore(todoEl, todoListEl.firstChild);
+  }
+
+  function getTodoItemElement(todo) {
+    // Creaate todo item root element ('list-item')
+    const todoEl = document.createElement("div");
+    todoEl.className = "list-item";
+
+    // Create checkbox with todo text ('checkbox-container')
+    const checkboxEl = document.createElement("label");
+    checkboxEl.className = "checkbox-container";
```

Comment:
Why you just not use the checkboxEl.className since you already define it here as const ?

```
checkboxEl.innerHTML = `
    <label class="${checkboxEl.className}">
        <input type="checkbox" />
        <span class="list-item-name">${todo}</span>
    </label>
    `;
```

Reply:
Thanks for the catch. Yes, obviously this is innerHTML should be without `<label .../>` wrapper. Will change later today!